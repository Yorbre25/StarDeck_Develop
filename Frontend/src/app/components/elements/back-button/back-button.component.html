<button type="submit" class="button-text" (click)="goBack()"> <mat-icon>keyboard_backspace</mat-icon></button>