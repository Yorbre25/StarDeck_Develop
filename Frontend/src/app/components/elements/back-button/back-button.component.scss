.button-text{
    margin: 20px;
    font-size: 0.2em;
    
}