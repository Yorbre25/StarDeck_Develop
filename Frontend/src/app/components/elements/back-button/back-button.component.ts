import { Component } from '@angular/core';
import { Router } from '@angular/router';
import { Location } from '@angular/common';

@Component({
  selector: 'app-back-button',
  templateUrl: './back-button.component.html',
  styleUrls: ['./back-button.component.scss']
})
export class BackButtonComponent {
  constructor(private router: Router, private location: Location) {}

  goBack() {
    if (this.location.path() === '') {
      this.router.navigate(['/home']);
    } else {
      this.location.back();
    }
  }
}
