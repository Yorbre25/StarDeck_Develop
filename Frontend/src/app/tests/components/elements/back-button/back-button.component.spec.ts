import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BackButtonComponent } from '../../../../components/elements/back-button/back-button.component';

describe('BackButtonComponent', () => {
  let component: BackButtonComponent;
  let fixture: ComponentFixture<BackButtonComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ BackButtonComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(BackButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('match with snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
