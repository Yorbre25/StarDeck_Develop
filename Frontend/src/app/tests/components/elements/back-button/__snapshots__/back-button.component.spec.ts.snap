// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButtonComponent match with snapshot 1`] = `
<div
  id="root1"
>
  <button
    class="button-text"
    type="submit"
  >
    <mat-icon>
      keyboard_backspace
    </mat-icon>
  </button>
</div>
`;
